server/src/main/scala/io/delta/common/SnapshotChecker.scala

```diff
@@ -0,0 +1,138 @@
+   * tablePropertiesWithDisabledValues, because clientReaderFeatures is empty for parquet format
+   * sharing rpcs, which won't filter out any properties in tablePropertiesWithDisabledValues.
+   *
+   * @param configuration The "configuration" field of a delta Metadata, which contains the
+   *                      properties of the table.
+   * @param tableVersion  The table vesrion associated with the configuration.
+   * @param clientReaderFeatures The set of reader features supported by the delta sharing client
+   * @param flagReaderFeatures The set of reader features allowed by the flag, used to enabling
+   *                           flags for providers to share tables with specific reader features.
+   */
+  def assertTableProperties(
+                             isProviderRpc: Boolean,
+                             configuration: Map[String, String],
+                             tableVersion: Option[Long],
+                             clientReaderFeatures: Set[String],
+                             flagReaderFeatures: Set[String] = Set.empty): Unit = {
+    // An unsupported table property can be supported if it is in part of the client supported
+    // table features.
+    def propertySupportedByClientOrFlag(property: String): Boolean = {
+      // TODO: @pranavsuku-db add deletionvector logic
+      if (property == DeltaAction.columnMappingProperty.property) {
+        ColumnMappingTableFeature.isInSet(clientReaderFeatures) ||
+          ColumnMappingTableFeature.isInSet(flagReaderFeatures)
+      } else {
+        // We should not reject any other table properties as they can contain arbitrary keys.
+        true
+      }
+    }
+    val unsupportedPropertiesByClient = DeltaAction.tablePropertiesWithDisabledValues
+      .flatMap {
+        case pr @ DeltaAction.PropertyAllowedValues(property, allowedValues)
+          if !propertySupportedByClientOrFlag(property) =>
+          configuration.get(property).filterNot(allowedValues.contains(_)).map(_ => pr)
+        case _ => None
+      }
+    if (unsupportedPropertiesByClient.nonEmpty) {
+      throw new DeltaSharingUnsupportedOperationException("Unsupported Delta Table Features")
+    }
+  }
+
+  def getTableFeaturesFromProperties(properties: Map[String, String]): Set[String] =
+    properties.keys
+      .filter(_.contains("delta.feature."))
+      .map { featureProp =>
+        val startIndx = "delta.feature.".length
+        featureProp.substring(startIndx)
+      }
+      .toSet
+
+  /**
+   * Assert all table features are supported by both the server and the client.
+   *
+   * This will NOT fail parquet format requests on tables with advanced features, because
+   * features like DV will persist in protocol.readerFeatures even if it's turned off by
+   * setting table properties. We rely on assertTableProperties to fail the request if
+   * the table properties are actually enabled.
+   */
+  def assertTableFeatures(
```

Comment:
similarly we need to add test for these functions, otherwise we can remove them.

Reply:
I think assertTableFeatures is not useful and here is why I think it can be removed:

For this function there is a comment 

> Assert all table features are supported by both the server and the client.
>   This will NOT fail parquet format requests on tables with advanced features, because
>    features like DV will persist in protocol.readerFeatures even if it's turned off by
>    setting table properties. We rely on assertTableProperties to fail the request if
>    the table properties are actually enabled.

assertTableProperties checks if features are not supported by the client. The assertTableFeatures is mainly checking if something is not handled by the server, but since we are using kernel, column mapping and deletion vectors will be supported by the server. The tests I wrote in the most recent commit go through assertTableProperties and fail there since the client cannot support them.